salt/states/archive.py

```diff
@@ -1304,7 +1315,7 @@ def extracted(name,
                 ret['changes']['directories_created'] = [name]
             ret['changes']['extracted_files'] = files
             ret['comment'] = '{0} extracted to {1}'.format(source_match, name)
-            if not keep:
+            if not source_is_local and not keep:
```

Comment:
Am I reading this right that the default behavior now is to remove the local cached copy of a remove archive? It looks like the docs say the default for `keep` is supposed to be `True` but here it looks like the default is `False`.

I noticed a large increase in network activity and time when I updated some minions recently and tracked down the source to `archive.extracted`. The source files are no longer being stored in the minion cache and are being downloaded to the minions each time highstate is run.

Could this be the cause of the change in behavior?

Reply:
@ntrepid8 The default behavior prior to 2017.7.3 would have been to remove the cached source archive. This was changed to keep the cached source in https://github.com/saltstack/salt/pull/43518 (backported for the 2017.7.3 release in https://github.com/saltstack/salt/pull/43681).

Reply:
:+1: